digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporter.scala

```diff
@@ -110,26 +119,73 @@ class TrafficSignCsvImporter extends CsvDataImporterOperations {
   }
 
   private def verifyValueCode(parameterName: String, parameterValue: String): ParsedAssetRow = {
-    if(parameterValue.forall(_.isDigit) && TRTrafficSignType.apply(parameterValue.toInt).source.contains("CSVimport")){
+    if(parameterValue.forall(_.isDigit) && TrafficSignType.applyTRValue(parameterValue.toInt).source.contains("CSVimport")){
       (Nil, List(AssetProperty(columnName = codeValueFieldMappings(parameterName), value = parameterValue.toInt)))
     }else{
       (List(parameterName), Nil)
     }
   }
 
-  def tryToInt(propertyValue: String ) = {
+  def getRightRoadLinkUsingBearing(assetBearing: Option[Int], assetCoordinates: Point): (Seq[VVHRoadlink], Seq[RoadLink]) = {
+    val toleranceInDegrees = 25
+    val roadLinks = roadLinkService.getClosestRoadlinkForCarTrafficFromVVH(userProvider.getCurrentUser(), assetCoordinates)
+
+    assetBearing match {
+      case Some(aBearing) if roadLinks.nonEmpty =>
+        val enrichedRoadLinks = roadLinkService.enrichRoadLinksFromVVH(roadLinks)
+
+        val filteredEnrichedRoadLinks =
+          enrichedRoadLinks.filter { roadLink =>
+            val aTrafficDirection = TrafficDirection.apply(trafficSignService.getTrafficSignValidityDirection(assetCoordinates, roadLink.geometry))
+
+            val mValue = GeometryUtils.calculateLinearReferenceFromPoint(assetCoordinates, roadLink.geometry)
+            val roadLinkBearing = GeometryUtils.calculateBearing(roadLink.geometry, Some(mValue), Some(roadLink.length) )
+
+            if (roadLink.trafficDirection == TrafficDirection.BothDirections) {
+              val reverseRoadLinkBearing =
+                if (roadLinkBearing - 180 < 0) {
+                  roadLinkBearing + 180
+                } else {
+                  roadLinkBearing - 180
+                }
+
+              Math.abs(aBearing - roadLinkBearing) <= toleranceInDegrees ||
+                Math.abs(aBearing - reverseRoadLinkBearing) <= toleranceInDegrees
+            } else {
+              Math.abs(aBearing - roadLinkBearing) <= toleranceInDegrees &&
+                (roadLink.trafficDirection == aTrafficDirection || aTrafficDirection == TrafficDirection.BothDirections)
```

Comment:
can `aTrafficDirection == TrafficDirection.BothDirections` be possible?

Reply:
During the tests, I get some examples of that but can be tested again to be sure.